clippy_lints/src/methods/unnecessary_fallible_conversions.rs

```diff
@@ -63,24 +64,42 @@ fn check<'tcx>(
                 // so that can be machine-applicable
 
                 (
+                    self_ty,
+                    other_ty,
                     "into()",
                     primary_span.with_hi(unwrap_span.hi()),
                     Applicability::MachineApplicable,
                 )
             },
-            FunctionKind::TryFromFunction => ("From::from", primary_span, Applicability::Unspecified),
-            FunctionKind::TryIntoFunction => ("Into::into", primary_span, Applicability::Unspecified),
-            FunctionKind::TryIntoMethod => ("into", primary_span, Applicability::Unspecified),
+            FunctionKind::TryFromFunction => (
+                other_ty,
+                self_ty,
```

Comment:
`other_ty` and `self_ty` and swapped in this arm

Reply:
This is on purpose, otherwise the datatypes in the note are reversed (e.g. converting `u64` to `u32` ...)